### ISSUE TITLE
Display a badge on "New products"

### ISSUE THREAD

**Issue by xC4M** [CrocoBlock/suggestions#3663]
I've been trying to show a "New!" badge on each new product in an ecommerce, but I haven't achieved it yet.
I was using Query Builder to display in a listing grid all new products with a "New!" badge, but when the product is on the Archive Shop Page (where all products are in), there is no way to display the "New!" badge on those new products. Likewise, I tried setting the dynamic visibility for it and testing with custom queries and so on...

I contacted the support team, and they suggested me to create a date field (using Meta Box) where I can duplicate the date and display the product's badge based on the date that I set in there.

Could you please add a way we could display custom badges?, specially in this kind of situation where customers must see which product is the new one, they have to identify them easily and that's why must be added a practical way to allow us including this kind of badge.

**Comment by 79ho3ein** [comment]
If I am not mistaken, a request similar to your request has already been submitted : #2680

Yes, this feature is really needed. Of course for the jet woobuilder plugin. I hope this feature will be added to the plugin soon. It is very practical.

**Comment by nreljed** [comment]
I have encountered this problem in the past and I have long thought about all the possibilities
I did not find better than this snippet-based solution
you can already start with that and modify it to your needs

```
/* Show / Hide badge if post is new (30 days) */
function newly_posted() {
global $post;
$now = date('U'); $published = get_the_time('U');
$new = false;
if( $now-$published  <= 30*24*60*60 ) $new = true;
return $new;
}

/* Shortcode for wodpress posts */
add_shortcode( 'new-post', function () {

if( newly_posted() ) 
	
	$out = '<span>New</span>';
	return $out;
} );
```
![Capture d’écran 2021-07-17 005055](https://user-images.githubusercontent.com/82481592/126018619-84860a38-884b-4e4a-a14b-6756a29e2564.png)

it displays new badge if the post or product is less than 30 days from the date of publication

place the [new-post] shortcode where you want it and of course css to customize it

like that if a dev passes by here it will speak directly to him , and understand what to do to adapt it differently with the plugin
or validate that there is no better, simpler solution

**Comment by UraraReika** [comment]
Hello @xC4M and @nreljed ! 

You can use this code snippet as an example https://gist.github.com/UraraReika/5e944f13d166ec9457698922272197ce and add your custom macros to all badges widgets in JetWooBuilder like this https://i.gyazo.com/85c3b91b9e384043075225a5e50b34d8.png.

**Comment by UraraReika** [comment]
I merged this issue with #2680, which is on the same subject so we could deal with it more efficiently.
